protocols/grpc/agent.proto

```diff
@@ -170,29 +174,29 @@ message Route {
 	string device = 3;
 }
 
-enum UpdateType {
-	None = 0;
-	AddIP = 1;
-	RemoveIP = 2;
-	Name = 4;
-	MTU = 8;
-}
-
 message AddInterfaceRequest {
-	Interface interface = 1;
+	string interface_id = 1;
+	Interface interface = 2;
 }
 
 message RemoveInterfaceRequest {
-	string name = 1;
+	string interface_id = 1;
 }
 
+// interface_id MUST NOT empty
+// all the other fields should be updated when it is set
 message UpdateInterfaceRequest {
-	UpdateType type = 1;
-	Interface interface = 2;
+	string interface_id = 1;
```

Comment:
Where would that `interface_id` come from? Is the runtime supposed to generate it?
Why do we need it when we have an interface name already?

Reply:
yes the runtime is supposed to generate it.
I'm afraid interface name is not always reliable.
I hope every resource has its ID which is generated/controlled by runtime or even upper layer.

Reply:
@laijs we used the macaddress as the identifier in clear containers as that is reliable.

Reply:
@laijs I really hope the runtime does not need to create, store and manage IDs when there already are reliable ways to identify resources. In the network case, I think the MAC address is a reliable ID, as @mcastelino says.

Reply:
I have same doubt, we don't need an interface_id. MAC address should be good enough.

Reply:
Actually I think a `name` is also unique and reliable ID, we can't have same `name` for two network interfaces.